packages/block-library/src/site-logo/editor.scss

```diff
@@ -77,9 +80,80 @@
 			justify-content: center;
 		}
 
-		// Hide drag and drop text.
+		// Hide items.
+		.components-placeholder__label,
```

Comment:
Similarly we can pass through a prop to not render the label, and we can remove this.